cmd/posix-list-dir_windows.go

```diff
@@ -29,6 +29,66 @@ func readDir(dirPath string) (entries []string, err error) {
 	return readDirN(dirPath, -1)
 }
 
+// readDir applies the filter function on each entries at dirPath, doesn't recurse into
+// the directory itself.
+func readDirFilterFn(dirPath string, filter func(name string, typ os.FileMode) error) error {
+	d, err := os.Open(dirPath)
+	if err != nil {
+		// File is really not found.
+		if os.IsNotExist(err) {
+			return errFileNotFound
+		}
+
+		// File path cannot be verified since one of the parents is a file.
+		if strings.Contains(err.Error(), "not a directory") {
+			return errFileNotFound
+		}
+		return err
+	}
+	defer d.Close()
+
+	st, err := d.Stat()
+	if err != nil {
+		return err
+	}
+	// Not a directory return error.
+	if !st.IsDir() {
+		return errFileAccessDenied
+	}
+
+	data := &syscall.Win32finddata{}
+
+	for {
+		e := syscall.FindNextFile(syscall.Handle(d.Fd()), data)
+		if e != nil {
+			if e == syscall.ERROR_NO_MORE_FILES {
+				break
+			} else {
+				err = &os.PathError{
+					Op:   "FindNextFile",
+					Path: dirPath,
+					Err:  e,
+				}
+				return err
+			}
+		}
+		name := syscall.UTF16ToString(data.FileName[0:])
+		if name == "" || name == "." || name == ".." { // Useless names
+			continue
+		}
+		if data.FileAttributes&syscall.FILE_ATTRIBUTE_REPARSE_POINT != 0 {
+			continue
+		}
+		if err = filter(name, typ); err == errSkipFile {
+			// We continue further filtering.
```

Comment:
ok, so this is why I was confused above.

If `errSkipFile` is returned we continue, if nil is returned we do not????

That seems to me to be the opposite of what makes sense. Maybe don't use `errSkipFile ` and make nil continue operation and an `errAbortCrawl` (or something like that) stop the listing.

Add that information to the documentation.